geoapps/grid_creation/application.py

```diff
@@ -0,0 +1,383 @@
+            bottom_padding
+        )
+        depth_core, expansion_fact = float(depth_core), float(expansion_fact)
+
+        trigger = callback_context.triggered[0]["prop_id"].split(".")[0]
+        update_dict = {}
+        if trigger == "upload":
+            if (filename.endswith("ui.json") or filename.endswith(".geoh5")) and (
+                contents is not None
+            ):
+                if filename.endswith(".ui.json"):
+                    update_dict.update(self.update_from_ui_json(contents, param_list))
+                    ws = update_dict["geoh5"]
+                elif filename.endswith(".geoh5"):
+                    content_type, content_string = contents.split(",")
+                    decoded = io.BytesIO(base64.b64decode(content_string))
+                    ws = Workspace(decoded)
+                update_dict.update(self.update_object_options(ws, "objects"))
+                update_dict.update(self.update_object_options(ws, "xy_reference"))
+                update_dict["filename"] = None
+                update_dict["contents"] = None
+            else:
+                print("Uploaded file must be a workspace or ui.json.")
+        elif trigger + "_name" in param_list:
+            update_dict[trigger + "_name"] = locals()[trigger]
+        elif trigger != "":
+            update_dict[trigger] = locals()[trigger]
+
+        self.update_param_dict(update_dict)
+        outputs = self.get_outputs(param_list, update_dict)
+
+        return outputs
+
+    def create_block_model(self, n_clicks):
+        # self.params should be up to date whenever create_block_model is called.
+        param_dict = self.params.to_dict()
+        temp_geoh5 = f"BlockModel_{time():.0f}.geoh5"
+
+        # Get output path.
+        if self.params.live_link:
+            if self.params.monitoring_directory is not None and os.path.exists(
+                os.path.abspath(self.params.monitoring_directory)
+            ):
+                output_path = self.params.monitoring_directory
+            else:
+                print("Invalid monitoring directory path")
+                return []
+        else:
+            output_path = os.path.dirname(self.params.geoh5.h5file)
+
+        # Get output workspace.
+        ws, self.params.live_link = self.get_output_workspace(
+            self.params.live_link, output_path, temp_geoh5
+        )
+        with ws as workspace:
+            # Put entities in output workspace.
+            param_dict["geoh5"] = workspace
+            for key, value in param_dict.items():
+                if isinstance(value, ObjectBase):
+                    param_dict[key] = value.copy(parent=workspace, copy_children=True)
+
+            # Write output uijson.
+            ifile = InputFile(
+                ui_json=self.params.input_file.ui_json,
+                validation_options={"disabled": True},
+            )
+            new_params = GridCreationParams(input_file=ifile, **param_dict)
```

Comment:
Call without the input_file argument. That way params will have a clean input file